### PR TITLE
Add ECMAScript module support

### LINKED ISSUE #89: typescript synthetic import
when using emoji-regex with es style imports such as
```ts
import emojiRegex from 'emoji-regex/es2015/RGI_Emoji';
```
typescript gives the following error:
```
This module is declared with using 'export =', and can only be used with a default import when using the 'allowSyntheticDefaultImports' flag.
```

looking at the [typescript documentation for default exports](https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html#default-exports) it looks like
```ts
export = emojiRegex;
```
should be replaced by 
```ts
export default emojiRegex;
```
inside each `.d.ts` file
I tried making this change manually inside my node_modules folder and it seems to resolve the issue

note: this is a regression because it used to work in v9.2.0 but it doesn't work in v9.2.2

### PR DESCRIPTION
**Pull request by nstringham**
This PR solves the same problem as #37 but in a different way

this PR also fixes #89

this PR allows users to do
```js
import emojiRegex from 'emoji-regex';
```
or
```js
const { default: emojiRegex } = await import('emoji-regex');
```
or
```js
const emojiRegex = require('emoji-regex');
```

it does this by adding index.mjs which is like index.js except that is an es module instead of a commonjs module

### CODE REVIEW

**Comment by chihabeeddine** [comment]
is there a reason why this pull request still open ?

File: script/build.js [hunk 1]
```diff
@@ -5,3 +5,8 @@ const pattern = fs.readFileSync('./node_modules/emoji-test-regex-pattern/dist/la
 const output = input.replace('<% pattern %>', pattern) + '\n';
 
 fs.writeFileSync('./index.js', output);
+
+const inputModule = fs.readFileSync('./src/index.mjs', 'utf8').toString().trim();
+const outputModule = inputModule.replace("<% pattern %>", pattern) + "\n";
+
+fs.writeFileSync("./index.mjs", outputModule);
```
Comment (mathiasbynens) [inline script/build.js:12]: To match the surrounding code style:

```suggestion
fs.writeFileSync('./index.mjs', outputModule);
```

File: src/index.mjs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default () =>{
```
Comment (mathiasbynens) [inline src/index.mjs:1]:
```suggestion
export default () => {
```

File: test/tests.js [hunk 3]
```diff
@@ -93,4 +93,9 @@ describe('regex', () => {
 		assert(/\\u\{/.test(regexSource) === false);
 	});
 
+	it('can be imported as an ecmascript module', async () => {
```
Comment (mathiasbynens) [inline test/tests.js:96]:
```suggestion
	it('can be imported as a JavaScript module', async () => {
```

File: index.mjs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export default () =>{
```
Comment (mathiasbynens) [inline index.mjs:1]:
```suggestion
export default () => {
```

**Review by mathiasbynens — CHANGES_REQUESTED** [review]

**Review by mathiasbynens — APPROVED** [review]

**Comment by mathiasbynens** [comment]
Thanks!

**Comment by himanshugogroup** [comment]
Still getting error->>> `TypeError: (0 , emoji_regex_1.default) is not a function`
For
```
    import  emojiRegex from 'emoji-regex';
    const emgRegex = emojiRegex(); <-- Failing here
```

**Comment by nstringham** [comment]
@himanshugogroup can you share a code example to reproduce your error? It's likely that this error is caused by a misconfigured build system. Are you using Typescript?

**Pull request merged**